### PR TITLE
Add `pyhalcyon` recepi

### PR DESCRIPTION
**Pull request by relastle**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/pyhalcyon/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+about:
+  home: The package home page
+  license: GPLv3
+  summary: "Halcyon: an accurate basecaller exploiting an encoder-decoder model with monotonic attention"
+  doc_url: https://github.com/relastle/halcyon
+  dev_url: https://github.com/relastle/halcyon
+
+extra:
+  recipe-maintainers:
+    - your-github-id-here
```
Comment (bgruening) [inline recipes/pyhalcyon/meta.yaml:77]: Please enter your GitHub name here

File: recipes/pyhalcyon/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    - pytest ==6.0.1
+    - flake8 ==3.8.3
+    - autopep8 ==1.5.4
+    - pyflakes ==2.2.0
+    - coverage ==5.2.1
+    - isort ==5.4.2
+    - poetry
+
+about:
+  home: The package home page
```
Comment (bgruening) [inline recipes/pyhalcyon/meta.yaml:69]: Pleas fill this out

File: recipes/pyhalcyon/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+build:
+  number: 0
+  entry_points:
+    - halcyon = halcyon.console:main
+  script: "{{ PYTHON }} -m pip install . -vv"
+  noarch: python
+
+requirements:
+  host:
```
Comment (bgruening) [inline recipes/pyhalcyon/meta.yaml:20]: Try to remove everything but pip and python

**Review by bgruening — COMMENTED** [review]
Thanks! Small review inline.

File: recipes/pyhalcyon/meta.yaml (hunk 1 above)
Comment (relastle) [reply recipes/pyhalcyon/meta.yaml]: Fixed in 689c7e1

File: recipes/pyhalcyon/meta.yaml (hunk 2 above)
Comment (relastle) [reply recipes/pyhalcyon/meta.yaml]: Fixed in d15b17a

File: recipes/pyhalcyon/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - pip
+    - python ==3.7.4
```
Comment (bgruening) [inline recipes/pyhalcyon/meta.yaml:23]: you should not pin python that strict, does this tool work with python3 in general?

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**